logging/telemetry.go

```
@@ -71,28 +72,64 @@ func makeTelemetryState(cfg TelemetryConfig, hookFactory hookFactory) (*telemetr
 	return telemetry, nil
 }
 
+// ReadTelemetryConfigOrDefault reads telemetry config from file or defaults if no config file found.
+func ReadTelemetryConfigOrDefault(dataDir *string, genesisID string) (cfg TelemetryConfig, err error) {
+	err = nil
+	if dataDir != nil && *dataDir != "" {
+		configPath := filepath.Join(*dataDir, TelemetryConfigFilename)
+		cfg, err = LoadTelemetryConfig(configPath)
+	}
+	if err != nil && os.IsNotExist(err) {
+		var configPath string
+		configPath, err = config.GetConfigFilePath(TelemetryConfigFilename)
+		if err != nil {
+			cfg = createTelemetryConfig()
+			return
+		}
+		cfg, err = LoadTelemetryConfig(configPath)
+	}
+	if err != nil {
```

Why are we swallowing the error here?